services/movie.py

```diff
@@ -1,26 +1,24 @@
-from typing import Optional
-
 from django.db.models import QuerySet
 
 from db.models import Movie
 
 
 def get_movies(
-        genres_ids: Optional[list[int]] = None,
-        actors_ids: Optional[list[int]] = None
+        genres_ids: list[int] | None = None,
+        actors_ids: list[int] | None = None
 ) -> QuerySet:
     query_set = Movie.objects.all()
     if genres_ids and actors_ids:
-        return query_set.filter(
+        query_set = query_set.filter(
             genres__id__in=genres_ids,
             actors__id__in=actors_ids
         )
-    if genres_ids:
-        return query_set.filter(
+    elif genres_ids:
+        query_set = query_set.filter(
             genres__id__in=genres_ids
         )
-    if actors_ids:
-        return query_set.filter(
+    elif actors_ids:
+        query_set = query_set.filter(
             actors__id__in=actors_ids
         )
     return query_set

```

Comment:
you didn't get it.
Let's remove the first if and leave the last to as "if", not "elif". In this case if both parameters are passed, python will get inside each if statement and appropriate filter will be added

Reply:
I got it after I'd commited. Now everything should be ok